### PR TITLE
Add "edge" mode to mx.pad

### LINKED ISSUE #1284: [Feature Request] mx.pad supports the "edge" padding mode.
The function `pad` in [numpy](https://numpy.org/doc/stable/reference/generated/numpy.pad.html) supports "edge" padding mode which will pads with the edge values of array.

### PR DESCRIPTION
**Pull request by barronalex**
Resolves #1284

### CODE REVIEW

File: mlx/ops.h [hunk 1]
```diff
@@ -256,30 +256,36 @@ array moveaxis(
     int destination,
     StreamOrDevice s = {});
 
+enum PaddingMode { Constant, Edge };
```
Comment (awni) [inline mlx/ops.h:259]: Maybe use an `enum class` here instead of a `enum`

Comment (awni) [reply mlx/ops.h:259]: On the other hand we tend to match types from Python API to C++ API (I'm not sure that's the best choice tbh), but for the sake of consistency maybe simpler to have it be a string at the `ops.h` level and scope the enum to the Primitive itself.

I think (?) it also makes it easier for code binding ops.h (i.e. MLX C) to not have to add the new enum type.

E.g. like `indexing` for `meshgrid`, `ord` for `norm`.

Comment (barronalex) [reply mlx/ops.h]: Sounds good! I changed it to just be a string.

**Review by awni — APPROVED** [review]
Looks great to me, thanks for adding that!

**Pull request merged**